pensieve/analysis.py

```diff
@@ -116,13 +123,13 @@ def run(self, experiment: experimenter.Experiment, current_date: datetime, dry_r
         # todo additional experiment specific metrics from Experimenter
         sql = exp.build_query(self.STANDARD_METRICS, last_window_limits, "normandy", None)
 
-        if self.bq_context is None:
-            self.bq_context = BigQueryContext(project_id=self.project, dataset_id=self.dataset)
-
         if dry_run:
             self.logger.info("Not executing query for %s; dry run", experiment.slug)
             return
 
+        if self.bq_context is None:
+            self.bq_context = BigQueryContext(project_id=self.project, dataset_id=self.dataset)
```

Comment:
Potentially, in future tests we might want to mock some of the behaviour when it comes to accessing BigQuery since the CI doesn't have permissions.

Reply:
:+1: